File: WebApplication/src/modules/student/models/education-basic.dto.ts
import { InstitutionDto } from "src/infrastructure/models/institution.dto";
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";

export class EducationBasicDto {
  id: number;
  institution: InstitutionDto;
  speciality: NomenclatureDto;
  educationalQualification: NomenclatureDto;
  educationalForm: NomenclatureDto;
  status: NomenclatureDto;
  researchArea: NomenclatureDto;
  duration: number;

  graduationDate: Date;
  interruptionReason: string;
}

File: StudentCredit.Application/Students/Dtos/EducationBasicDto.cs
using StudentCredit.Infrastructure.Helpers.Dtos;

namespace StudentCredit.Application.Students.Dtos
{
    public class EducationBasicDto
    {
        public int Id { get; set; }
        public InstitutionNomenclatureDto Institution { get; set; }

        public NomenclatureDto Speciality { get; set; }

        public NomenclatureDto EducationalQualification { get; set; }

        public NomenclatureDto EducationalForm { get; set; }

        public NomenclatureDto Status { get; set; }

        public NomenclatureDto ResearchArea { get; set; }
        public decimal? Duration { get; set; }

        //Дата на дипломиране
        public DateTime? GraduationDate { get; set; }

        //Ако е със статус "Прекъснал"
        public string InterruptionReason { get; set; }

    }
}
